crates/sui-json-rpc/src/name_service.rs

```diff
@@ -60,6 +60,14 @@ impl Domain {
             type_params: vec![],
         }
     }
+
+    /// Derive the parent domain for a given domain
+    /// E.g. `test.example.sui` -> `example.sui`
+    pub fn parent(&self) -> Domain {
```

Comment:
Do we want this to be Option<Domain> to gracefully handle the case where we have Domain that is just a TLD? Also if this is "example.sui" then this presently will return "sui", is that the desired behavior?

Reply:
So, on another PR for subdomains, I am actually limitting the creation of `Domain` only to SLD (len >=2) (since we won't ever have a TLD being an actual domain). So there, there's not really a reason to guard it. I am just repeating the code snippet here as I haven't merged the other PR 😅  

The `example.sui` -> `sui` is not a problem + adds simple SLD querying for names by looking up `sui`'s children. Can probably make sense if we decide to index `.move` names in the same DB in the future!